ExampleMod/Content/NPCs/ExampleGlobalNPC.cs

```diff
@@ -27,17 +31,53 @@ public class ExampleGlobalNPC : GlobalNPC
 		}
 
 		//If the merchant has been hit by a player, they will double their sell price
-		public override void SetupShop(int type, Chest shop, ref int nextSlot) {
-			if (Main.LocalPlayer.talkNPC == -1 || !Main.npc[Main.LocalPlayer.talkNPC].GetGlobalNPC<ExampleGlobalNPC>().HasBeenHitByPlayer) {
+		public override void ModifyActiveShop(NPC npc, string shopName, Item[] items) {
+			if (!npc.GetGlobalNPC<ExampleGlobalNPC>().HasBeenHitByPlayer) {
 				return;
 			}
 
-			foreach (Item item in shop.item) {
+			foreach (Item item in items) {
 				int value = item.shopCustomPrice ?? item.value;
 				item.shopCustomPrice = value * 2;
 			}
 		}
 
+		// Example of adding new items with complex conditions in the Merchant shop.
+		public override void ModifyShop(NPCShop shop) {
+			// Style 1 check for application
+			if (shop.Name != NPCShopDatabase.GetNPCShopName(NPCID.Merchant, "Shop"))
+				return;
+
+			// Style 2 check for application
+			if (shop.Name != "Terraria/Merchant/Shop")
+				return;
+
+			// Adding ExampleTorch to Merchant, with condition being sold only during daytime. Have it appear just after Torch
+			shop.InsertAfter(ItemID.Torch, ModContent.ItemType<Items.Placeable.ExampleTorch>(), NPCShop.Condition.TimeDay);
+
+			// Hiding Copper Pickaxe and Copper Axe. They will never appear in Merchant shop anymore
+			// However, this approach may fail if item doesn't exists in shop.
+			shop.GetEntry(ItemID.CopperAxe).Disable();
+
+			// Safer approach for disabling item
+			if (shop.TryGetEntry(ItemID.CopperPickaxe, out NPCShop.Entry entry)) {
+				entry.Disable();
+			}
+
+			// Adding new Condition to Blue Flare. Now it will appear just if player carries a Flare Gun in their inventory AND is in Snow biome
+			shop.GetEntry(ItemID.BlueFlare).AddCondition(NPCShop.Condition.InSnowBiome);
+
+			// Creaate a condition for later use.
+			var redPotCondition = new NPCShop.Condition(NetworkText.FromKey("Mods.ExampleMod.ShopConditions.RedPotCondition"), () => !NPCShop.Condition.HappyWindyDay.IsAvailable() || !NPCShop.Condition.HappyEnough.IsAvailable());
```

Comment:
Should go next to the usage. 
Comment should say 'custom condition, opposite of conditions for `ExampleItem` above'